grpc/cmd/server/main.go

```diff
@@ -29,11 +29,26 @@ func main() {
 	}
 	defer lis.Close()
 
-	go runGRPC(lis)
-	runHTTP(clientAddr)
+	//this channel will be used to halt
```

Comment:
You would have to talk me into it, as this seems like a much more complicated approach than the one I've had before. Why would I choose an 8-liner over my two lines?